functors_task_1/student/src/impl.cpp

```diff
@@ -1,2 +1,37 @@
 #include "impl.h"
+#include <array>
+#include <algorithm>
 
+BeerBrand BeerOrganizer::operator()() noexcept {
+    if (m_number==m_max){
+        m_number = m_min+1;
+    }
+    return static_cast<BeerBrand>(m_number++);
+}
+
+BeerOrganizer::BeerOrganizer():m_number(m_min+1) {
+    static_assert(m_max-m_min>1, "You have only None and Max in enum!");
+}
+
+bool isExpensiveBeer(const BeerBrand brand) {
+    constexpr static std::array<BeerBrand, 2> expensiveBeers =
```

Comment:
I suppose that is overengineering, please simplify with plain logical expression